contracts/Core/AssetManager.sol

```diff
@@ -306,6 +282,124 @@ contract AssetManager is ACL, AssetStorage {
         );
     }
 
+    function addPendingCollections() 
+        external
+        onlyRole(parameters.getAssetConfirmerHash()) 
+    {
+        if(numPendingCollections != 0)
+        {
+            
+            uint256 temp = numPendingCollections;
+            for(uint8 i = 1; i <= temp; i++){
+                uint256 currentEpoch = parameters.getEpoch();
+                numAssets = numAssets + 1;
+
+                collections[numAssets].id = numAssets; 
+                collections[numAssets].name = pendingCollections[i].name; 
+                collections[numAssets].aggregationMethod = pendingCollections[i].aggregationMethod; 
+                collections[numAssets].epoch = currentEpoch;
+                collections[numAssets].creator = pendingCollections[i].creator;
+                collections[numAssets].repeat = pendingCollections[i].repeat;
+                collections[numAssets].assetType = uint256(assetTypes.Collection);
+                collections[numAssets].active = true;
+                uint256[] memory jobIDs = pendingCollections[i].jobIDs;
+                for(uint256 j = 0; j < jobIDs.length; j++){
```

Comment:
Code complexity O(m*n)
anything above O(n) should be avoided.

Reply:
Been done this way to mark each jobID that is a part of the collection which is being done by a mapping.